src/main/java/org/dcsa/ebl/service/impl/ShippingInstructionServiceImpl.java

```diff
@@ -215,6 +212,49 @@
             siTO -> Mono.just(shippingInstructionMapper.dtoToShippingInstructionResponseTO(siTO)));
   }
 
+  Mono<List<Booking>> validateDocumentStatusOnBooking(ShippingInstructionTO shippingInstructionTO) {
+
+    List<String> carrierBookingReferences = new ArrayList<>();
+    if (shippingInstructionTO.getCarrierBookingReference() != null) {
+      carrierBookingReferences.add(shippingInstructionTO.getCarrierBookingReference());
+    } else {
+      carrierBookingReferences =
+          shippingInstructionTO.getShipmentEquipments().stream()
+              .flatMap(x -> x.getCargoItems().stream().map(CargoItemTO::getCarrierBookingReference))
+              .collect(Collectors.toList());
+    }
+    carrierBookingReferences = carrierBookingReferences.stream().distinct().collect(Collectors.toList());
```

Comment:
```suggestion
    List<String> carrierBookingReferences;
    if (shippingInstructionTO.getCarrierBookingReference() != null) {
      carrierBookingReferences = Collections.singletonList(shippingInstructionTO.getCarrierBookingReference());
    } else {
      carrierBookingReferences =
          shippingInstructionTO.getShipmentEquipments().stream()
              .flatMap(x -> x.getCargoItems().stream().map(CargoItemTO::getCarrierBookingReference))
              .distinct()
              .collect(Collectors.toList());
    }
```

Reply:
NB: Written by hand directly in Github - please review it manually (it might need additional imports)